app/assets/javascripts/components/modal-dialogue.js

```diff
@@ -0,0 +1,88 @@
+function ModalDialogue ($module) {
+  this.$module = $module
+  this.$dialogBox = $module.querySelector('.app-c-modal-dialogue__box')
+  this.$closeButton = $module.querySelector('.app-c-modal-dialogue__close-button')
+  this.$body = document.querySelector('body')
+}
+
+ModalDialogue.prototype.init = function () {
+  if (!this.$module) {
+    return
+  }
+
+  this.$module.boundOpen = this.open.bind(this)
+  this.$module.boundClose = this.close.bind(this)
+  this.$module.boundKeyDown = this.handleKeyDown.bind(this)
+
+  var $triggerElement = document.querySelector('[data-toggle="modal"][data-target="#' + this.$module.id + '"]')
+  if ($triggerElement) {
+    $triggerElement.addEventListener('click', this.$module.boundOpen)
+  }
+
+  if (this.$closeButton) {
+    this.$closeButton.addEventListener('click', this.$module.boundClose)
+  }
+}
+
+ModalDialogue.prototype.open = function (event) {
+  if (event) {
+    event.preventDefault()
+  }
+  this.$body.classList.add('app-o-template__body--modal')
```

Comment:
Nope not isolated, but I don't see other way forward. The body needs to know if a modal is being displayed or not.

Reply:
Putting aside the blur effect for a moment (which looks good but isn't essential) why does the body need to know? Is it something to do with preventing scrolling while the modal is open?

Reply:
Yep, to disable scrolling the main page

Reply:
I see. Yeah, doesn't seem possible to do that without altering the body. Why is that a requirement though?

Reply:
There is a design requirement to prevent the main body from scrolling when the modal is open. The reason behind this is because the scroll on body element is automatically triggered (bubbles) when you reach the bottom/top scroll of the modal (best seen is long modal example).